File: mikkel_agent/helpers/negotiation_progress.py
from geniusweb.progress.ProgressRounds import ProgressRounds


class NegotiationProgress:

    def __init__(self, progress: ProgressRounds):
        self._progress = progress

    def advance(self):
        self._progress = self._progress.advance()

    def get_progress(self) -> float:
        return self._progress.get(0)

    def get_object(self) -> ProgressRounds:
        return self._progress

File: areaparty/src/main/java/geniusweb/exampleparties/areaparty/NegotiationProgress.java
package geniusweb.exampleparties.areaparty;

import geniusweb.progress.Progress;
import geniusweb.progress.ProgressRounds;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class NegotiationProgress {

    private Progress progress;

    public double getProgress() {
        if (this.progress == null) {
            return 0;
        }
        return this.progress.get(System.currentTimeMillis());
    }

    public void advance() {
        this.progress = this.getProgressRounds().advance();
    }

    private ProgressRounds getProgressRounds() {
        return (ProgressRounds) progress;
    }
}
